### DISCUSSION TITLE
Graph/plotting issues with CSV file and Grafana 8

### DISCUSSION THREAD

**Discussion by Kybeer** [grafana/grafana-infinity-datasource#112] [General]
I'm using Infinity 0.7.5 and Grafana 8.0.6 to visualize CSV files. Getting the CSV data as Table view in Grafana is no problem. But **I'm facing issues when trying to "graph/plot" it.**

I cannot even reproduce the example from https://sriramajeyam.com/grafana-infinity-datasource/wiki/csv/ in my Grafana setup:

Infinity help example for CSV plotting (from link above):

![image](https://user-images.githubusercontent.com/57593336/128032280-7677f8c2-75d9-470e-8e80-f988e989d6cb.png)

When I try with the same settings, there's no **graph being plotted**:

![image](https://user-images.githubusercontent.com/57593336/128032356-d4906865-b2e1-4693-a6dc-1971b7485cc2.png)

If I use **table** view (but otherwise use the same settings) the **date gets invalid**:

![image](https://user-images.githubusercontent.com/57593336/128032900-1fab8d66-c0e1-4261-98c5-6f21a4e8a5a5.png)

If I don't use any **columns** on the **table** view, the data is displayed correct:

![image](https://user-images.githubusercontent.com/57593336/128033237-f0e4c01e-61e6-4991-a79a-f1cc5dd5e8c5.png)

Anyone with clues on how to get the date / graph plotting correct?

br
johan

**Comment by yesoreyeram** [comment]
Sorry you had issues. Not sure what is wrong with your query. See if the below recording helps.

https://user-images.githubusercontent.com/153843/128044628-628aa5ca-b431-4873-a235-6e785f45e8c9.mov

**Reply by Kybeer** [reply]
Thanks a lot for providing this help!

When I follow your video I get **"invalid date"** at 0:23s mark (see red marking in the screen shot). Perhaps it's some time setting issue on my end ...?

I also see our plugin options aren't exactly the same. I'm missing a few buttons (green arrow markings in the screen shot)

![image](https://user-images.githubusercontent.com/57593336/128051309-c5665149-dd40-4679-97e0-2d0e849efbf3.png)


vs this from the video:

![image](https://user-images.githubusercontent.com/57593336/128051578-0dae2ccf-d879-4895-82a4-66d0da995725.png)

**Reply by Kybeer** [reply]
If I use **Timestamp** the dates gets invalid:

![image](https://user-images.githubusercontent.com/57593336/128053925-0738c765-2bc7-4d1d-bf6c-a8cd632512f6.png)

If I use **String** it's OK, but I cannot "graph" it

![image](https://user-images.githubusercontent.com/57593336/128054052-a81b0d08-6e1d-4053-a11d-7b4ee7b3ac4c.png)

![image](https://user-images.githubusercontent.com/57593336/128054085-9179aa62-0f7a-4eb7-a1ea-56288b633cbf.png)

**Reply by Kybeer** [reply]
Perhaps it's the tailing "UTC" in the created_at column that breaks it on my end for what ever reason...?

**Reply by Kybeer** [reply]
I tested a CSV file without tailing "UTC" in the time stamp column, and it works. 

Don't know the reason behind it all, perhaps some local time zone settings.
